gexpect_test.go

```diff
@@ -236,3 +236,43 @@ func TestRegexTimeoutWithOutput(t *testing.T) {
 		t.Fatalf("Didn't find %v in output: %v", searchPattern, out)
 	}
 }
+
+func TestRegexFindNoExcessBytes(t *testing.T) {
+	t.Logf("Testing Regular Expressions returning output with no excess strings")
+	repeats := 100
+	seqCmd := fmt.Sprintf("`seq 1 %d`", repeats)
```

Comment:
Do you mean something like matching the `a \d+ b` regexp in "a 1 b a 2 b a 3 c" string? Sure, I can test it.